### ISSUE TITLE
GU0009 should not warn when in expression trees

### ISSUE THREAD

**Issue by JohanLarsson** [JohanLarsson/Gu.Analyzers#107]
```c#
namespace RoslynSandbox
{
    using System;
    using System.Linq.Expressions;

    internal class Foo
    {
        public void Bar()
        {
            Meh(() => Id(true)); // GU0009 should not warn here
        }

        internal static void Meh(Expression<Action> meh)
        {
        }

        private static bool Id(bool self) => self;
    }
}
```
